docs/api/python/text/text.md

```diff
@@ -120,73 +112,137 @@ by specifying their indices (2 and 1) and the `glossary.idx_to_vec` in
+Suppose that we want to build indices for the 2 most frequent keys in `counter` with the unknown
+token representation '<UnK>' and a reserved token '<pad>'.
 
 ```python
->>> fasttext_simple = text.embedding.TokenEmbedding.create('fasttext',
-...     pretrained_file_name='wiki.simple.vec')
+>>> my_vocab = text.vocab.Vocabulary(counter, most_freq_count=2, unknown_token='<UnK>', 
+...     reserved_tokens=['<pad>'])
 
 ```
 
-Suppose that we have a simple text data set in the string format. We can count
-word frequency in the data set.
+We can access properties such as `token_to_idx` (mapping tokens to indices), `idx_to_token` (mapping
+indices to tokens), `vec_len` (length of each embedding vector), and `unknown_token` (representation
+of any unknown token) and `reserved_tokens`.
+
+
+```python
+>>> my_vocab.token_to_idx
+{'<UnK>': 0, '<pad>': 1, 'world': 2, 'hello': 3}
+>>> my_vocab.idx_to_token
+['<UnK>', '<pad>', 'world', 'hello']
+>>> my_vocab.unknown_token
+'<UnK>'
+>>> my_vocab.reserved_tokens
+['<pad>']
+>>> len(my_vocab)
+4
+```
+
+Besides the specified unknown token '<UnK>' and reserved_token '<pad>' are indexed, the 2 most
+frequent words 'world' and 'hello' are also indexed.
+
+
+
+
+## Text token embedding
+
+To load token embeddings from an externally hosted pre-trained token embedding file, such as those
+of GloVe and FastText, use
+[`_embedding.create(embedding_name, pretrained_file_name)`](#mxnet.contrib.text.embedding.create).
+
+To get all the available `embedding_name` and `pretrained_file_name`, use
+[`_embedding.get_embedding_and_pretrained_file_names()`](#mxnet.contrib.text.embedding.get_embedding_and_pretrained_file_names).
+
+```python
+>>> text.embedding.get_embedding_and_pretrained_file_names()
+{'glove': ['glove.42B.300d.txt', 'glove.6B.50d.txt', 'glove.6B.100d.txt', ...],
+'fasttext': ['wiki.en.vec', 'wiki.simple.vec', 'wiki.zh.vec', ...]}
+
+```
+
+Alternatively, to load embedding vectors from a custom pre-trained text token
+embedding file, use [`CustomEmbedding`](#mxnet.contrib.text.embedding.CustomEmbedding).
+
+Moreover, to load composite embedding vectors, such as to concatenate embedding vectors,
+use [`CompositeEmbedding`](#mxnet.contrib.text.embedding.CompositeEmbedding).
+
+The indexed tokens in a text token embedding may come from a vocabulary or from the loaded embedding
+vectors. In the former case, only the indexed tokens in a vocabulary are associated with the loaded
+embedding vectors, such as loaded from a pre-trained token embedding file. In the later case, all
+the tokens from the loaded embedding vectors, such as loaded from a pre-trained token embedding
+file, are taken as the indexed tokens of the embedding.
+
+
+```eval_rst
+.. currentmodule:: mxnet.contrib.text.embedding
+.. autosummary::
+    :nosignatures:
+
+    register
+    create
+    get_embedding_and_pretrained_file_names
+    GloVe
+    FastText
+    CustomEmbedding
+    CompositeEmbedding
```

Comment:
Seems that this isn't in the right module

Reply:
Resolved.